### ISSUE TITLE
Umlauts on Windows with american mac keyboard.

### ISSUE THREAD

**Issue by kirschem** [rbreaves/kinto#622]
**Is your feature request related to a problem? Please describe.**
Using an american keyboard layout on macOS, I can type "umlauts" (ü,ä,ö) by typing OPT+U and then u, e or a.
This does not work with the current setup on Windows.

**Describe the solution you'd like**
Having the same behavior as described above.

**Describe alternatives you've considered**
It seems like the windows way of doing that would be AltGr + a/u/o, but that does not seem to work either (AltGr should be opt on mac, right?)

**Additional context**

**Comment by kirschem** [comment]
Added a PR: https://github.com/rbreaves/kinto/pull/647

**Comment by rbreaves** [comment]
Thanks for the PR! Keeping this open till I replicate this on the Linux side.

**Comment by RedBearAK** [comment]
@rbreaves @kirschem 

Am I mistaken in thinking that this should be easily expandable, or at least replicatable, to work for pretty much all the other accented keys that are easily accessible in macOS with Option+Key1, Key2? Like the grave/acute accents, the enye, the "c" with the dangly bit, and so forth. Code seems pretty straightforward. 

On the other hand, I have this code already in my `kinto.ahk` since I just reinstalled Kinto on Windows 11, but it doesn't appear to be working as expected. It waits for the second key to be pressed, as it should, but then instead of a character with an umlaut I get a capital "A" with a tilde over it, and then a second character. Same goes for the other keys, just with different second characters. Like this: [ Ã¶ Ã– Ã¤ Ã„ Ã¼ Ãœ ]. Hope that comes out the way I'm seeing in my browser. 

I tried changing the "u" with umlaut to send "xyz" instead, and that works. It inserts "xyz" in Notepad when I type Opt+u, u. But the umlaut characters in the AHK config file seem to be incompatible with my (standard American, as far as I know) character set or something. I don't really know how that works in Windows. But to be universal it seems this might need to get AHK to output the characters in a different way. 

I'll keep playing with it for a bit.

**Comment by RedBearAK** [comment]
Well, after a number of different things didn't work, this works, using the numpad key names to create an Alt+Nums keystroke for the special character. But, it only works ONCE, then I have to quit Notepad or it will never work again. Same thing happens when I just try to type Alt+0246 or something with or without Kinto running. It works once and then quits working until I quit the affected application. So, probably nothing to do with Kinto or AHK, but something funky with my Win11. Seems to be a widespread problem with no known solution. At least the basic keystroke send works. On a Windows machine not affected by this weird Alt+Code bug, this kind of thing should work permanently. 

These characters all appear to be correct, but they are listed as "Latin [ Small | Capital ] Letter [ O | A | U ] With Diaeresis", which appears to be the same symbol over the letters. I can't find any characters in the character map app described as "with umlaut". 

```
#IfWinNotActive ahk_group remotes
    $!u::Goto, ActivateUmlautModifier
    $!s::Send, ß

    ActivateUmlautModifier:
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 B
    if UserInput = o
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad6}{Alt up} ; ö (Alt Code: Alt+0246)
    else if UserInput = O
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad4}{Alt up} ; Ö (Alt Code: Alt+0214)
    else if UserInput = a
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad8}{Alt up} ; ä (Alt Code: Alt+0228)
    else if UserInput = A
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad6}{Alt up} ; Ä (Alt Code: Alt+0196)
    else if UserInput = u
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad2}{Alt up} ; ü (Alt Code: Alt+0252)
    else if UserInput = U
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad0}{Alt up} ; Ü (Alt Code: Alt+0220)
    else
        Send, %UserInput%
    return
#If
```

**Comment by RedBearAK** [comment]
@rbreaves @kirschem 

Weird. That problem I was having with the Alt Codes might actually be restricted to Notepad. The shortcuts seem to be working fine in other applications. Firefox, VSCode, Notepad++. üöäÜÖÄ. It just keeps working. 

This should work for the small "sharp s" or eszett. Works for me. 

```
    $!s::Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad3}{Alt up} ; ß (Alt Code: Alt+0223)
```

**Comment by kirschem** [comment]
I had the same problem, then I‘ve also tried with the alt codes and experienced the same issue in notepad.

I went back to the first solution and found out that the ahk file needs to be saved with utf8+BOM encoding.

**Comment by RedBearAK** [comment]
> I had the same problem, then I‘ve also tried with the alt codes and experienced the same issue in notepad.
> 
> I went back to the first solution and found out that the ahk file needs to be saved with utf8+BOM encoding.
> 
> Still undecided what’s the better solution. It’s strange that the alt codes don’t work in the regular notepad consistently.

@rbreaves @kirschem 

There, that’s what I was thinking. A character encoding mismatch between the file and the OS. So VSCode knows what character to show me as I look at the config file, but when it’s pasted into another application it comes out as a pair of ISO-8859 characters or whatever Windows is still using internally. 

It would certainly be advisable to make sure all text files are Unicode these days, but I will argue that the most robust solution to getting AHK to reliably insert the correct character will be to use a technique like this that abstracts the character away from the encoding of the AHK file. Because there is no way to predict what the user might do to the file, like copy new bits into an older customized config file that would still be in the non-Unicode encoding. Or save the Unicode file in a way that changes the encoding. 

Unless there is some other better method for AHK to output a Unicode character correctly regardless of the encoding of the config file, I think this Alt Code method will be the wisest way to go. As long as Windows itself and the specific application hold up their part of the deal (unlike Notepad.exe) and allow the use of the Alt Codes manually, this technique should always work for everyone. Plus I think it’s good to actually put the relevant Alt Codes in the comments as I’ve done here. For reference.

**Comment by RedBearAK** [comment]
@rbreaves @kirschem 

Found a couple of good references for all the Option and Option+Shift symbols and special characters available on the Mac keyboard. 

This one is mainly symbols, seems to not show all the individual accented keys. But shows character names for all the symbols on the list, which is really helpful for searching the Character Map. 

https://altcodesguru.com/mac-alt-key-codes.html

Here is another reference, with a different presentation but lacking the character names. Does a better job showing the whole selection of accented characters, like the Umlaut characters. 

https://slcr.wsu.edu/help-pages/keyboards-osx/

Looks like there are at least several dozen in total on the entire keyboard, between Option+Key-then-Key, Option+Key, and Option+Shift+Key combinations. 

Here's a reference that seems to contain a complete list of 523 characters with names, but that includes a lot of Option+alphanumeric codes similar to the Windows Alt+0nnn codes. Split up into different categories. 

https://www.webnots.com/4-ways-to-type-accented-letters-in-mac/

I'll just start with the accented characters and some common symbols like the Euro. We'll see how far I can get.

**Comment by RedBearAK** [comment]
@joshgoebel 

>Can't we do that here already with just a simple nested mapping? 

I don't know, can we? I've never actually tried to implement a nested mapping with `xkeysnail`. If that's already do-able, I'll try to tackle it. The sequences are like Option+dead_key, then the key to be accented, resulting in an accented character. 

I had thought that the nested stuff would only work on the output side, is that not correct? That part of the Readme always confused me. 

One of the things that I'm making part of the AHK implementation is the ability to hit Escape to cancel the dead keys sequence. I don't know how easily that can be done in `xkeysnail`. 

And the way the output is working in AHK on Windows is I'm specifically outputting the keystrokes necessary to do the "Alt Code" for each special character. Like Alt(down)+(numpad)0,2,3,2+Alt(up), etc. That bypasses an issue with AHK not putting out the right character if the character encoding of the config file wasn't UTF-8. 

There must be a similar way, if not identical, to use UTF or Alt Codes to send special characters to Linux apps. I'm not that familiar just yet if there is a universal way to do that in Linux that doesn't require the activation of a "Compose" key option or something. 

The other thing I should be able to do with AHK is to disable/enable the scheme with a specific keyboard shortcut. I don't know how to do that with `xkeysnail`. Would need to set up a variable and be able to toggle the variable with the keyboard shortcut, and have an entire (large) section of the config file be deactivated if the variable is "off".

**Comment by joshgoebel** [comment]
> The sequences are like Option+dead_key, then the key to be accented, 

```
K("Alt-dead_key") : {
  K("u"): # output the accented U here
}
```

>There must be a similar way, if not identical, to use UTF or Alt Codes to send special characters to Linux apps. 

Dunno but if it can be done with just normal keyboard strokes, then we can probably do it.

> disable/enable the scheme with a specific keyboard shortcut. I don't know how to do that with xkeysnail.

Now that might be a more difficult ask... it it wasn't burdensome why would we need to turn it on and off?

**Comment by joshgoebel** [comment]
> On Xorg the default Compose Key is ⇧ Shift+AltGr

https://en.wikipedia.org/wiki/Compose_key#Common_compose_combinations

So this behavior is already build into linux if one has (or remaps) a compose key...

**Comment by RedBearAK** [comment]
@joshgoebel 

I get an error at the curly brace, attempting to use something basically identical to the example in the Readme. 

```
# Keybindings for GNOME Text Editor
define_keymap(re.compile("gnome-text-editor", re.IGNORECASE),{
    K("M-e"):   {
        K("a"): K("Shift-A"),
        K("e"): K("Shift-E"),
    }
```

I don't know if the Kinto patch somehow breaks this or what, but it's not working for me. I think that's why I never really messed with it before.

**Comment by joshgoebel** [comment]
> I don't know if the Kinto patch breaks this or what

Likely.  That is the thing people were complaining about IIRC - multi key macros.

**Comment by joshgoebel** [comment]
Why do you need the Kinto version, only the Alt-tab behavior?

**Comment by RedBearAK** [comment]
@joshgoebel 

>Now that might be a more difficult ask... it it wasn't burdensome why would we need to turn it on and off?

The more of the keyboard you implement for Option-key character input, the more you overlap with attempting to get to menu items in a Windows or Linux environment where Alt is often used to access those menu items. So on a Mac, Option+S will do an Eszett/beta character from German. But if that is active, you can't do Alt+S to get to the "Selection" menu in VS Code. Just as an example. 

This isn't a problem on macOS because Alt isn't used to access the menu items in the same way. Menu items just have a shortcut that is almost always based on the Cmd key, and used directly to access the item rather than open the "menu". 

I am trying to do a conditional bypass in AHK where if the user attempts the same shortcut again it will just pass through, but that isn't working just yet. And that would only apply to the "dead keys" shortcuts (used for the common accented characters), not the direct-entry shortcuts that instantly send a specific character. 

>Why do you need the Kinto version again, only the Alt-tab sticky behavior?

Yes, the "held keys" patch. Without that, the shortcut fires immediately when pressed, and it's impossible to Alt+Tab through the task switcher repeatedly before releasing Alt, or Alt+Grave through more than two application windows. So you're stuck switching between just two windows or apps, no matter how many of either are actually open. 

Have to take off for a while. Will get back into this later.

**Comment by RedBearAK** [comment]
@joshgoebel 

If the nested shortcuts ability could be fixed under Kinto, that would be HUGE.

**Comment by RedBearAK** [comment]
@joshgoebel 

So, this is the thing that kept me from using the special character entry method on Linux in the past, it seems to be dependent on some settings in your DE. 

I was able to get this Compose shortcut to work, but only after setting my "Compose Key" in GNOME settings to the `Menu` key. If it's set to anything else the shortcut just brings up a context menu. 

```
# Keybindings for GNOME Text Editor
define_keymap(re.compile("gnome-text-editor", re.IGNORECASE),{
    K("M-s"):                [K("Compose"),K("s"),K("s")],
```

I have to look into this further to see if there is a method that would work independently of how the user has their Compose key configured in the DE. 

But the Compose combos themselves are remarkably logical, even if they are different from the way Apple does it. Should be less annoying to implement than the Windows Alt Codes method.

**Comment by RedBearAK** [comment]
@joshgoebel 

This may be the only universal way to do special character entry on Linux: 

```
Other operating systems
The Alt key method does not work on Chrome OS, Mac OS, Linux or other [operating systems](https://en.wikipedia.org/wiki/Operating_system) and there is no readily-accessible evidence of interest in replicating it, due to its including the 1980s [IBM PC](https://en.wikipedia.org/wiki/IBM_PC) character encoding as part of its definition. However, numeric entry of Unicode characters is possible in most Unix or Unix-like OSs by typing Ctrl+⇧ Shift+U, (release) then the hex number, then the space bar or enter key.
```

This would be a pain since the Unicode number often doesn't match the older Alt Code number.

**Comment by joshgoebel** [comment]
> I was able to get this Compose shortcut to work, but only after setting my "Compose Key" in GNOME settings to the Menu key. 

https://wiki.archlinux.org/title/Xorg/Keyboard_configuration#Configuring_compose_key

There isn't a compose key on most keyboards, so you have to map it... but you could make it with `modmap` inside xkeysnail as well, you don't need Gnome... but you will need to find a real key to map to compose.

**Comment by joshgoebel** [comment]
>but only after setting my "Compose Key" in GNOME settings to the Menu key. 

I'm not sure why you'd have to do this if we're sending the literal Compose key via the virtual keyboard...  I might have to muck around... I use Awesome WM so there is no big DE in my setup at all.

**Comment by RedBearAK** [comment]
@joshgoebel 

>There isn't a compose key on most keyboards, so you have to map it... but you could make it with modmap inside xkeysnail as well, you don't need Gnome... but you will need to find a real key to map to compose.

Yeah, I don't really understand how that works. I have no "real" PgUp/PgDn/Home/End keys on my Acer laptop(s), just the ones on the Numpad, which failed to be recognized by any GTK application. I was able to implement a fix for that without a `modmap`: 

https://github.com/rbreaves/kinto/pull/499

>Weird, acts like a right click here in Chrome...

Yep, that's what it does to me, until I set the chosen Compose key to "Menu".

**Comment by joshgoebel** [comment]
The keys coming in at the low-level are definitely Compose-S-S (according to evtest)... but it also doesn't work for me until I run `setxkbmap -option compose:menu` AFTER we take over... if I run that before we take over it seems to have no effect.

I think it's something weird with X because if you fire up `xev` it will report the key as `menu`.

**Comment by RedBearAK** [comment]
@joshgoebel 

Actually, Compose, S, S would make the capital Eszett character, which is apparently not widely used. Compose, s, s would make the small Eszett. Gotta be careful with the case when talking about this stuff. Capital AE ligature will require a Shift-A and Shift-E with the way `xkeysnail` does things. 

>doesn't work for me until I run setxkbmap -option compose:menu AFTER we take over

I have no hope of being able to troubleshoot that issue myself, without learning a lot more about how the low-level devices work (starting from zero). But if you could figure that out it so that Compose could be invoked reliably, it would be a heck of a lot easier to implement the Compose combos than the Unicode input sequences.

**Comment by RedBearAK** [comment]
@joshgoebel @kirschem @rbreaves 

Updated code. Supports the entire number keys row and a significant portion of the letter keys. Still working on getting a "passthrough" to work correctly. 

There is a problem with a few characters like "Not Equal To" and "fi" and "fl" ligatures. No working Alt Codes. 

```autohotkey

; ###############################################################################################################
; ###   Special character insertion like Apple/macOS Option key methods, mapping to Windows Alt+Codes method
; ###   Planned: Common symbols available with Option+key, common accented keys with Option+Key1, then Key2
; ###############################################################################################################
#IfWinNotActive ahk_group remotes
    ; Paste this partial Alt+0 code after double colons to save typing: 
    ; Send, {Alt down}{Numpad0}{Numpad}{Numpad}{Numpad}{Alt up} 

    ; Shortcut to activate Option key special character scheme
    ^+!o::MsgBox, 0, WHOOPSIE!, Not working yet.`n`nWill activate Option key special characeter entry scheme.`n`nMay interfere with some main menu bar shortcuts.`n`nTry tapping Alt first, then the letter of the menu.`n`n


    ; ######################################################################################
    ; ###   Individual Option+key and Option+Shift+key symbols/characters
    ; ######################################################################################

    ; ######   NUMBER KEYS ROW   ######

    ; Spanish opening/starting Exclamation Point symbol: ¡ (Alt Code: Alt+0161)
    !1::Send {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad1}{Alt up}

    ; Trademark symbol: ™ (Alt+0153)
    !2::Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad3}{Alt up}
    ; Euro currency symbol: € (Alt+0128)
    !+2::Send, {Alt down}{Numpad0}{Numpad1}{Numpad2}{Numpad8}{Alt up}

    ; British Pound currency symbol: £ (Alt+0163)
    !3::Send, {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad3}{Alt up}
    ; UNKNOWN SYMBOL: some kind of left angled bracket
    ; !+3::Send, {Alt down}{Numpad0}{Numpad1}{Numpad2}{Numpad8}{Alt up}

    ; Cent currency symbol: ¢ (Alt+0162)
    !4::Send, {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad2}{Alt up}
    ; UNKNOWN SYMBOL: some kind of right angled bracket
    ; !+4::Send, {Alt down}{Numpad0}{Numpad1}{Numpad2}{Numpad8}{Alt up}

    ; Infinity symbol: ∞ (Alt+236) [no zero in front!]
    !5::Send, {Alt down}{Numpad2}{Numpad3}{Numpad6}{Alt up}
    ; fi latin small ligature: (U+FB01) (No Alt Code? Requires Unicode.)
    ; !+5::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; Section mark symbol: § (Alt+0167)
    !6::Send, {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad7}{Alt up}
    ; fl small ligature: (U+FB02) (No Alt Code? Requires Unicode.)
    ; !+6::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; Paragraph mark (Pilcrow) symbol: ¶ (Alt+0182)
    !7::Send, {Alt down}{Numpad0}{Numpad1}{Numpad8}{Numpad2}{Alt up}
    ; Double dagger (cross) symbol: ‡ (Alt+0135) [Simple dagger/cross: Alt+0134]
    !+7::Send, {Alt down}{Numpad0}{Numpad1}{Numpad3}{Numpad5}{Alt up}

    ; Bullet point symbol: • (Alt+0149) 
    !8::Send, {Alt down}{Numpad0}{Numpad1}{Numpad4}{Numpad9}{Alt up}
    ; Degree symbol: ° (Alt+0176)
    ; NOT degree symbol: Option+k: Overring circle diacritic accent
    ; NOT degree symbol: Option+0: Masculine Ordinal Indicator (Alt+167 or 0186)
    !+8::Send, {Alt down}{Numpad0}{Numpad1}{Numpad7}{Numpad6}{Alt up}

    ; Feminine Ordinal Indicator symbol: ª (Alt+0170)
    !9::Send, {Alt down}{Numpad0}{Numpad1}{Numpad7}{Numpad0}{Alt up}
    ; Middle Dot (interpunct/middot) symbol: · (Alt+0183) 
    !+9::Send, {Alt down}{Numpad0}{Numpad1}{Numpad8}{Numpad3}{Alt up}

    ; Masculine Ordinal Indicator symbol: º (Alt+0186)
    !0::Send, {Alt down}{Numpad0}{Numpad1}{Numpad8}{Numpad6}{Alt up}
    ; Single low-9 quotation mark (U+201A): ‚ (Alt+0130) 
    !+0::Send, {Alt down}{Numpad0}{Numpad1}{Numpad3}{Numpad0}{Alt up}

    ; En dash symbol: – (Alt+0150)
    !-::Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad0}{Alt up}
    ; Em dash symbol: — (Alt+0151)
    !+-::Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad1}{Alt up}

    ; Not Equal To symbol (U+2260): (Alt+8800) [This Alt Code doesn't work!]
    ; !=::Send, {Alt down}{Numpad8}{Numpad8}{Numpad0}{Numpad0}{Alt up}
    ; Just send "not equal" sequence as a shortcut.
    !=::Send, {!}{=}
    ; Plus Minus symbol: ± (Alt+0177)
    !+=::Send, {Alt down}{Numpad0}{Numpad1}{Numpad7}{Numpad7}{Alt up}


    ; ######   LETTER KEYS   ###### [ QWERTY order ]


    ; ##############
    ; ###   1st row: Tab-key row [ qwertyuiop[]\ ] [ QWERTYUIOP{}| ]


    ; Small oe (oethel) ligature: œ (Alt+0156)
    !q::Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad6}{Alt up}
    ; Capital OE (Oethel) ligature: Œ (Alt+0140)
    !+q::Send, {Alt down}{Numpad0}{Numpad1}{Numpad4}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!w::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; ###   Option+e is Acute accent, see Dead_Keys_Accent_Acute

    ; Registered symbol: ® (Alt Code: Alt+0174)
    $!r::Send, {Alt down}{Numpad0}{Numpad1}{Numpad7}{Numpad4}{Alt up}

    ; Simple dagger (cross) symbol: † (Alt+0134)
    !t::Send, {Alt down}{Numpad0}{Numpad1}{Numpad3}{Numpad4}{Alt up}

    ; Yen currency symbol: ¥ (Alt+0165)
    !y::Send, {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad5}{Alt up}

    ; ###   Option+u is Umlaut accent, see Dead_Keys_Accent_Umlaut

    ; ###   Option+i is Circumflex accent, see Dead_Keys_Accent_Circumflex

    ; Small slashed o: ø (Alt+0248)
    !o::Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad8}{Alt up}
    ; Capital slashed O: Ø (Alt+0216)
    !+o::Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad6}{Alt up}

    ; Pi symbol: π (Alt Code: Alt+227)
    $!p::Send, {Alt down}{Numpad2}{Numpad2}{Numpad7}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $![::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!]::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!\::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}


    ; ##############
    ; ###   2nd row: Caps Lock row [ asdfghjkl;' ] [ ASDFGHJKL:" ]


    ; Overring accented "a/A" used by Nordic, Greenlandic, Germanic languages
    ; 
    ; Small "a" with overring: å (Alt+0229)
    $!a::Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad9}{Alt up}
    ; Capital "A" with overring: Å (Alt+0197)
    $!+a::Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad7}{Alt up}

    ; Small German Eszett/beta (Sharfes/Sharp S): ß (Alt Code: Alt+0223)
    $!s::Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad3}{Alt up}
    ; Capital German Eszett/beta (Sharfes/Sharp S): [ There is no "capital" ß. ]
    $!+s::Send, {S}{S}

    ; (Alt Code: Alt+0000)
    ; $!d::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; florin currency symbol: ƒ (Alt Code: Alt+159)
    $!f::Send, {Alt down}{Numpad1}{Numpad5}{Numpad9}{Alt up}

    ; Option+g brings up Windows XBox Game Bar! 
    ; To remove/disable paste this text in Powershell (without quotes): 
    ; "Get-AppxPackage Microsoft.XboxGamingOverlay | Remove-AppxPackage"
    ; 
    ; Copyright symbol: © (Alt+0169)
    !g::Send, {Alt down}{Numpad0}{Numpad1}{Numpad6}{Numpad9}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!h::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!j::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!k::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!l::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $![semicolon]::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $![colon]::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; Small ae ligature: æ (Alt+0230) - SC028 is single-quote key
    !SC028::Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad0}{Alt up}
    ; Capital AE ligature: Æ (Alt+0198)
    !+SC028::Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad8}{Alt up}


    ; ##############
    ; ###   3rd row: Shift-Keys row [ zxcvbnm,./ ] [ ZXCVBNM<>? ]


    ; (Alt Code: Alt+0000)
    ; $!z::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; (Alt Code: Alt+0000)
    ; $!x::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; Cedilla, small: ç (Alt+0231)
    $!c::Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad1}{Alt up}
    ; Cedilla, capital: Ç (Alt+0199)
    $!+c::Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad9}{Alt up}

    ; ###   Option+v is Caron accent, see Dead_Keys_Accent_Caron

    ; (Alt Code: Alt+0000)
    ; $!b::Send, {Alt down}{Numpad0}{Numpad0}{Numpad0}{Numpad0}{Alt up}

    ; ###   Option+n is Tilde accent, see Dead_Keys_Accent_Tilde

    ; Micro (mu) symbol: µ (Alt Code: Alt+0181)
    $!m::Send, {Alt down}{Numpad0}{Numpad1}{Numpad8}{Numpad1}{Alt up}

    ; Less than or equal to symbol: ≤ (Alt Code: Alt+243)
    $!,::Send, {Alt down}{Numpad2}{Numpad4}{Numpad3}{Alt up}
    ; Greater than or equal to symbol: ≥ (Alt Code: Alt+242)
    $!.::Send, {Alt down}{Numpad2}{Numpad4}{Numpad2}{Alt up}

    ; Obelus/Division symbol: ÷ (Alt Code: Alt+0247)
    $!/::Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad7}{Alt up}
    ; Spanish inverted question mark: ¿ (Alt Code: Alt+0191)
    $!+/::Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad1}{Alt up}



    ; Small eth:  (Alt+0240) [only on ABC Extended keyboard]
    ; !t::Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad0}{Alt up}
    ; Capital Eth:  (Alt+0208) [only on ABC Extended keyboard]
    ; !+t::Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad8}{Alt up}

    ; Small thorn:  (Alt+0222) [only on ABC Extended keyboard]
    ; !t::Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad2}{Alt up}
    ; Capital Thorn:  (Alt+0254) [only on ABC Extended keyboard]
    ; !+t::Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad4}{Alt up}


    ; ######################################################################################
    ; ###   DEAD KEYS: Option+dead_key to choose the desired diacritical mark, 
    ; ###              then press key to enter accented character with that mark
    ; ######################################################################################
    
    ; Watch for Apple "dead keys" Option key accented character sequences
    ; Go to appropriate processor function to create accented character
    ; 
    ; Acute accent: Option+e, then key to accent
    $!e::Goto, Dead_Keys_Accent_Acute                           ; FINISHED
    ; Circumflex accent: Option+i, then key to accent
    $!v::Goto, Dead_Keys_Accent_Caron                           ; TODO
    ; Circumflex accent: Option+i, then key to accent
    $!i::Goto, Dead_Keys_Accent_Circumflex                      ; FINISHED
    ; Grave accent: Option+`, then key to accent
    $!SC029::Goto, Dead_Keys_Accent_Grave                       ; FINISHED
    ; Tilde accent: Option+n, then key to accent
    $!n::Goto, Dead_Keys_Accent_Tilde                           ; FINISHED
    ; Umlaut/Diaeresis accent: Option+u, then key to accent
    $!u::Goto, Dead_Keys_Accent_Umlaut                          ; FINISHED

    ; Reference for Alt Codes for foreign accentend characters: 
    ; https://usefulshortcuts.com/alt-codes/accents-alt-codes.php


    Dead_Keys_Accent_Acute:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Acute accent activated by Option+e (logical Alt+e)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !e
        Send, {Alt down}{Alt up}e
    else if UserInput = a
        ; á (Alt Code: Alt+0225)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad5}{Alt up}
    else if UserInput = A
        ; Á (Alt Code: Alt+0193)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad3}{Alt up}
    else if UserInput = e
        ; é (Alt Code: Alt+0233)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad3}{Alt up}
    else if UserInput = E
        ; É (Alt Code: Alt+0201)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad1}{Alt up}
    else if UserInput = i
        ; í (Alt Code: Alt+0237)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad7}{Alt up}
    else if UserInput = I
        ; Í (Alt Code: Alt+0205)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad5}{Alt up}
    else if UserInput = o
        ; ó (Alt Code: Alt+0243)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad3}{Alt up}
    else if UserInput = O
        ; Ó (Alt Code: Alt+0211)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad1}{Alt up}
    else if UserInput = u
        ; ú (Alt Code: Alt+0250)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad0}{Alt up}
    else if UserInput = U
        ; Ú (Alt Code: Alt+0218)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad8}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return


    Dead_Keys_Accent_Caron:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Caron accent activated by Option+v (logical Alt+v)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !v
        Send, {Alt down}{Alt up}v
    else if UserInput = s
        ; š (Alt Code: Alt+0154)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad4}{Alt up}
    else if UserInput = S
        ; Š (Alt Code: Alt+0138)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad3}{Numpad8}{Alt up}
    else if UserInput = z
        ; ž (Alt Code: Alt+0158)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad8}{Alt up}
    else if UserInput = Z
        ; Ž (Alt Code: Alt+0142)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad4}{Numpad2}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return


    Dead_Keys_Accent_Circumflex:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Circumflex accent activated by Option+i (logical Alt+i)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !i
        Send, {Alt down}{Alt up}i
    else if UserInput = a
        ; â (Alt Code: Alt+0226)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad6}{Alt up}
    else if UserInput = A
        ; Â (Alt Code: Alt+0194)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad4}{Alt up}
    else if UserInput = e
        ; ê (Alt Code: Alt+0234)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad4}{Alt up}
    else if UserInput = E
        ; Ê (Alt Code: Alt+0202)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad2}{Alt up}
    else if UserInput = i
        ; î (Alt Code: Alt+0238)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad8}{Alt up}
    else if UserInput = I
        ; Î (Alt Code: Alt+0206)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad6}{Alt up}
    else if UserInput = o
        ; ô (Alt Code: Alt+0244)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad4}{Alt up}
    else if UserInput = O
        ; Ô (Alt Code: Alt+0212)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad2}{Alt up}
    else if UserInput = u
        ; û (Alt Code: Alt+0251)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad1}{Alt up}
    else if UserInput = U
        ; Û (Alt Code: Alt+0219)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad9}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return


    Dead_Keys_Accent_Grave:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Grave accent activated by Option+` (Alt plus scan code SC029, or !SC029)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !SC029
        Send, {Alt down}{Alt up}{SC029}
    else if UserInput = a
        ; à (Alt Code: Alt+0224)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad4}{Alt up}
    else if UserInput = A
        ; À (Alt Code: Alt+0192)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad2}{Alt up}
    else if UserInput = e
        ; è (Alt Code: Alt+0232)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad2}{Alt up}
    else if UserInput = E
        ; È (Alt Code: Alt+0200)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad0}{Alt up}
    else if UserInput = i
        ; ì (Alt Code: Alt+0236)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad6}{Alt up}
    else if UserInput = I
        ; Ì (Alt Code: Alt+0204)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad4}{Alt up}
    else if UserInput = o
        ; ò (Alt Code: Alt+0242)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad2}{Alt up}
    else if UserInput = O
        ; Ò (Alt Code: Alt+0210)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad0}{Alt up}
    else if UserInput = u
        ; ù (Alt Code: Alt+0249)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad9}{Alt up}
    else if UserInput = U
        ; Ù (Alt Code: Alt+0217)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad7}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return


    Dead_Keys_Accent_Tilde:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Tilde accent activated by Option+n (logical Alt+n)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !n
        Send, {Alt down}{Alt up}n
    else if UserInput = a
        ; ã (Alt Code: Alt+0227)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad7}{Alt up}
    else if UserInput = A
        ; Ã (Alt Code: Alt+0195)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad5}{Alt up}
    else if UserInput = n
        ; ñ (Alt Code: Alt+0241)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad1}{Alt up}
    else if UserInput = N
        ; Ñ (Alt Code: Alt+0209)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad9}{Alt up}
    else if UserInput = o
        ; õ (Alt Code: Alt+0245)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad5}{Alt up}
    else if UserInput = O
        ; Õ (Alt Code: Alt+0213)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad3}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return


    Dead_Keys_Accent_Umlaut:
    ; Use Apple "dead keys" Option key method to attach accents to next character typed
    ; Umlaut/Diaeresis accent activated by Option+u (logical Alt+u)
    StringCaseSense, On
    ; watch next input string
	Input, UserInput, L1 
    ; Watch for Escape key, cancel dead keys sequence
    if UserInput = Esc
        Return
    ; Option/Alt key needed to access menus sometimes
    ; If user repeats same shortcut, just send to app unmodified.
    else if UserInput = !u
        Send, {Alt down}{Alt up}u
    else if UserInput = a
        ; ä (Alt Code: Alt+0228)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad8}{Alt up}
    else if UserInput = A
        ; Ä (Alt Code: Alt+0196)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad9}{Numpad6}{Alt up}
    else if UserInput = e
        ; ë (Alt Code: Alt+0235)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad5}{Alt up}
    else if UserInput = E
        ; Ë (Alt Code: Alt+0203)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad3}{Alt up}
    else if UserInput = i
        ; ï (Alt Code: Alt+0239)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad3}{Numpad9}{Alt up}
    else if UserInput = I
        ; Ï (Alt Code: Alt+0207)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad0}{Numpad7}{Alt up}
    else if UserInput = o
        ; ö (Alt Code: Alt+0246)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad4}{Numpad6}{Alt up}
    else if UserInput = O
        ; Ö (Alt Code: Alt+0214)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad1}{Numpad4}{Alt up}
    else if UserInput = u
        ; ü (Alt Code: Alt+0252)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad2}{Alt up}
    else if UserInput = U
        ; Ü (Alt Code: Alt+0220)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad2}{Numpad0}{Alt up}
    else if UserInput = y
        ; ÿ (Alt Code: Alt+0255)
        Send, {Alt down}{Numpad0}{Numpad2}{Numpad5}{Numpad5}{Alt up}
    else if UserInput = Y
        ; Ÿ (Alt Code: Alt+0159)
        Send, {Alt down}{Numpad0}{Numpad1}{Numpad5}{Numpad9}{Alt up}
    ; No relevant character to accent? Send input unmodified. 
    else
        Send, %UserInput%
    Return

#If
; ### END of special character insertion with Option(Alt) key
```

**Comment by joshgoebel** [comment]
@RedBearAK Do not be afraid to ping me if you really think I should be in the loop, but I have little interest in the Windows side of Kinto... I use only Mac and Linux and know nothing about the keymapper software being used on windows and what it can or can't do. :-)

**Comment by RedBearAK** [comment]
@joshgoebel 

Well, this is the macOS Option-key special character scheme, and I'll be using this as the basis for bringing the same thing to the Linux config, if and when that becomes possible with the Kinto branch. I'll tag you back in when it's done. 

I do hope to make all of this work in Linux. I guess the main thing stopping me at the moment is the broken nested shortcuts in the Kinto branch. 

Does your new branch have both the Kinto "held keys" ability, and nested shortcuts working?

**Comment by joshgoebel** [comment]
> and I'll be using this as the basis for bringing the same thing to the Linux config,

Ok, keep me in the loop as you feel reasonable, but mostly the config looks like gibberish to me...

> both the Kinto "held keys" ability, and nested shortcuts working?

Nested short-cuts have always worked.  Held keys half work.  I currently only works (goes sticky) for simple A -> B keymaps, like CMD -> Ctrl.  So:

- If you hit `CMD-tab` (and hold CMD) CMD->Ctrl is marked as sticky (and the release of control is bound to CMD)
- If you hit say `CMD-shift-tab` nothing magical happens.

But if you START with CMD-tab and then you start shifting it'll work because the CMD is sticky and it doesn't matter that that shift will be reasserted each time.  How many people start tabbing backwards?  Might be some wouldn't even notice....

But if there are other places that have `X-Y-tab => X-tab` or `X-tab => `X-Y-tab` those don't currently work.

**Comment by RedBearAK** [comment]
@joshgoebel 

> How many people start tabbing backwards? 

I use it quite a bit, actually. It's a handy way to reverse and loop back to the other end of the task switcher and grab an app that is otherwise at the "bottom" of the tab stack, (i.e., at the opposite end of the task switcher dialog). With a lot of different "apps" open it can be tedious to tab through the entire task switcher, even if it is a vastly simplified representation versus the type of task switcher that shows previews of **_all_** windows open in the workspace.

**Comment by joshgoebel** [comment]
I didn't say go backwards - I said **start** backwards.  I'm not sure I've ever started that way personally ... I first open the switcher to see the state - at least that's me.  Backwards works fine IF you lead with Cmd-tab.  (which the auto-sticky knows how to handle)

**Comment by RedBearAK** [comment]
@joshgoebel 

>I didn't say go backwards - I said start backwards

I see. But if you know where you are going, to an app that you just used Cmd+H to "hide", for instance, it would be perfectly logical to start with Cmd+Shift+Tab. Since you would know through experience that the hidden app would be shunted to the back end of the task switcher. So it's like, "Oops, I didn't mean to hide that," and Cmd+Shift+Tab to get right back to it. 

And that does seem to work just fine with my existing Kinto setup. Although I'm not certain if I've ever actually started that way. It would be reasonable for users to expect it to work, in my opinion.

**Comment by joshgoebel** [comment]
> . Since you would know through experience that the hidden app would be shunted to the back end of the task switcher.

You have way better memory/spatial awareness than I do. 💯 

> But if you know where you are going, 

Sure - I didn't say that no one ever does it or we shouldn't support it - just *in my experience* it's very strange (to start backwards) and I'd still wager that holds for a lot of other people. :)  Anyways my point was for "testing" purposes you could probably "get by" with how things stand now.... I pinged you on the thread about fully resolving this (properly), which is a bit harder.

**Comment by RedBearAK** [comment]
@rbreaves @kirschem @joshgoebel 

If anyone is interested, I submitted PR #719 to support the full selection of special characters from the standard US Apple keyboard (about 137 symbols, punctuation marks and accented characters). The scheme is disabled by default to avoid interfering with Alt and Alt+Shift shortcuts. It can be enabled/disabled from the tray icon menu or with the shortcut Shift+Option+Command+O (the letter "o"). 

This is only for the Windows version of Kinto using AutoHotKey. The Linux version will hopefully be forthcoming in the future.

**Comment by rbreaves** [comment]
This has been merged. Leaving open till we replicate this on the Linux side.

**Comment by joshgoebel** [comment]
_The Linux side may require first switching to keyszer._

**Comment by rbreaves** [comment]
> The Linux side may require first switching to keyszer.

I was wondering if you had some developments in this area already. And I will be looking to make this transition soon, would say this 4th weekend - but my gf may have other plans for me than me coding my 4th of july weekend away :p.

**Comment by joshgoebel** [comment]
> I was wondering if you had some developments in this area already. 

I think @RedBearAK has the entire shebang working... but I could be wrong...

**Comment by rbreaves** [comment]
He probably does lol 😂, I did see a thread about it too I think but I just didn't have the bandwidth at the time to participate much right then.

**Comment by RedBearAK** [comment]
@rbreaves 

> I was wondering if you had some developments in this area already.

`keyszer` has a patch branch that I initiated but @joshgoebel did at least 90% of the work on, which is currently able to deal with strings with a limited selection of ASCII printables, and Unicode, either as part of a string or with a separate function called directly from the config file. Technically there are now two or three different ways to do Unicode characters with that branch. And it's working quite nicely. Hopefully it can get merged into main before too long. 

I still need to figure out how to enable/disable an entire keymap or subsection of a keymap with a keyboard shortcut. It really needs to be inactive by default like it is on the Windows PR, since the scheme interferes with other Alt+key shortcuts while active. I designed it to be as easy as possible to turn on when you need it for character entry, and off when you don't. 

Was also thinking of adding a new icon at some point to show when the Option-key scheme is active. Something like red inside or outside the `k`.

**Comment by joshgoebel** [comment]
> Hopefully it can get merged into main before too long.

Yeah we should be able to do that soon then crack a new release.

**Comment by RedBearAK** [comment]
@rbreaves 

Had to make a new PR to fix up some typos and formatting inconsistencies with the Option-key stuff. PR #726. Hope that can get merged before the next release.
